source/docs/guide/src/calc.md

```diff
@@ -0,0 +1,76 @@
+# Structured Proofs by Calculation
+
+## Motivation
+
+Sometimes, you need to establish some relation `R` between two expressions, say,
+`a_1` and `a_n`, where it might be easier to do this in a series of steps, `a_1`
+to `a_2`, `a_2` to `a_3`, ... all the way to `a_n`. One might do this by just
+doing all the steps at once, but as mentioned in the section on
+[assert-by](./assert_by.md), a better approach might be to split it into a
+collection of restricted contexts. This is better, but still might not be ideal,
+since you need to repeat each of the intermediate expressions at each point.
+
+## `calc!`ulations, to Reduce Redundant Redundancy
+
+The `calc!` macro supports structured proofs through calculations.
```

Comment:
Can you add a simple example with concrete assertions here, similar to the one at the end?

Reply:
Thanks, added